---
title: "bacmethy"
author: "Liu Jihong"
date: "`r Sys.Date()`"
output: html_document
---

## Figure 1 Simulated Sequencing Signal for Different Bases

```{r pressure, echo=FALSE}

set.seed(123)
time_length <- 200 
bases <- c("A", "T", "C", "G")

signal_data <- data.frame(
  Time = 1:time_length,
  Signal = runif(time_length, min = 0, max = 100),  
  Base = sample(bases, time_length, replace = TRUE)  
)

signal_data$AdjustedSignal <- with(signal_data, {
  ifelse(Base == "A", Signal * runif(1, 0.8, 1.2),
         ifelse(Base == "T", Signal * runif(1, 0.8, 1.2),
                ifelse(Base == "C", Signal * runif(1, 1.5, 2.0),  
                       Signal * runif(1, 0.8, 1.2))))  
})
  
 
  library(ggplot2)
  p<-ggplot(signal_data, aes(x = Time, y = AdjustedSignal, color = Base)) +
    geom_line() +  
    scale_color_manual(values = c("A" = "green", "T" = "red", "C" = "blue", "G" = "yellow")) +
    labs(title = "Simulated Sequencing Signal for Different Bases",
         x = "Time (arbitrary units)", y = "Signal Strength") +
    theme_minimal()
  ggsave("Simulated Sequencing Signals.eps", p, width = 16, height = 9, device = 'eps')
  ggsave("Simulated Sequencing Signals.pdf", p, width = 16, height = 9)
  
  
```

## figure 1 heatmap

```{r}
library(pheatmap) 
library(vegan)
nafamily<-read.table("../data/nofamily.txt",sep = "\t",header = F)
colnames(nafamily)<-c("locus tag","name","product","TBCF","TBCF Î”CP")
nafamily<-nafamily[-1,]
rownames(nafamily)<-nafamily$name
data<-as.data.frame(lapply(nafamily[,c(4,5)], as.numeric))
data.2<-log(data)
rownames(data.2)<-rownames(nafamily)
pheatmap(data.2,border=F)
data.t<-t(data.2)
pheatmap(data.t)

postscript("heatmap.eps", width = 6, height = 6, horizontal = FALSE, onefile = FALSE)


pheatmap(mat)


dev.off()
```

## figure 1 scatter plot

```{r}
data <- read.csv("/Users/jihong/Documents/MED_PG/lab/02_methylation/12_pipeline/19_briefinginbioinformatics/24_iMeta_review/source data/Figure-1/Ecolim6A.csv")

ggplot(data, aes(x=Coverage, y=IdentificationQv, color=Fraction)) +
  geom_point() +  
  geom_hline(yintercept=40, linetype="dashed", color="red") +  
  scale_color_gradient(low="#BE8693", high="#9dafc4") +  
  labs(title="E. coli K12\nm6A", x="Coverage", y="IdentificationQv") +  
  theme_minimal() +  
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", size=16),  
    axis.title = element_text(size=14),  
    axis.text = element_text(size=12),  
    legend.position="none",  
    axis.line = element_line(colour = "black"), 
    axis.ticks = element_line(colour = "black") 
  )


ggsave("Ecoli_K12_m6A_plot.pdf", width=8, height=6)

ggsave("Ecoli_K12_m6A_plot.eps", device = "eps", width=8, height=6)
ggsave("Ecoli_K12_m6A_plot.ems", device = "ems", width=8, height=6)

```

